docs/tutorials/transformer.ipynb

```diff
@@ -1,2354 +1,2386 @@
+    "\n",
+    "plt.pcolormesh(pos_encoding, cmap='RdBu')\n",
+    "plt.ylabel('Depth')\n",
+    "plt.xlabel('Position')\n",
+    "plt.colorbar()\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "a_b4ou4TYqUN"
+   },
+   "source": [
+    "## Masking"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "s42Uydjkv0hF"
+   },
+   "source": [
+    "Mask all the pad tokens in the batch of sequence. It ensures that the model does not treat padding as the input. The mask indicates where pad value `0` is present: it outputs a `1` at those locations, and a `0` otherwise."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "U2i8-e1s8ti9"
+   },
+   "outputs": [],
+   "source": [
+    "def create_padding_mask(seq):\n",
+    "  seq = tf.cast(tf.math.equal(seq, 0), tf.float32)\n",
+    "\n",
+    "  # add extra dimensions to add the padding\n",
+    "  # to the attention logits.\n",
+    "  return seq[:, tf.newaxis, tf.newaxis, :]  # (batch_size, 1, 1, seq_len)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "A7BYeBCNvi7n"
+   },
+   "outputs": [],
+   "source": [
+    "x = tf.constant([[7, 6, 0, 0, 1], [1, 2, 3, 0, 0], [0, 0, 0, 4, 5]])\n",
+    "create_padding_mask(x)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "Z0hzukDBgVom"
+   },
+   "source": [
+    "The look-ahead mask is used to mask the future tokens in a sequence. In other words, the mask indicates which entries should not be used.\n",
+    "\n",
+    "This means that to predict the third token, only the first and second token will be used. Similarly to predict the fourth token, only the first, second and the third tokens will be used and so on."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "dVxS8OPI9uI0"
+   },
+   "outputs": [],
+   "source": [
+    "def create_look_ahead_mask(size):\n",
+    "  mask = 1 - tf.linalg.band_part(tf.ones((size, size)), -1, 0)\n",
+    "  return mask  # (seq_len, seq_len)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "yxKGuXxaBeeE"
+   },
+   "outputs": [],
+   "source": [
+    "x = tf.random.uniform((1, 3))\n",
+    "temp = create_look_ahead_mask(x.shape[1])\n",
+    "temp"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note: Take note that this mask indicates which entries to *not* attend to. The newly added MultiHeadAttention layer in Keras considers the invered mask, where 1 is a token to be attended to and 0 should be ignored. If you are using `tf.keras.layers.MultiHeadAttention` instead of the MultiHeadAttention layer in this tutorial defined below, use the following commented out method to generate the mask. You will need to `pip install tensorflow-probability` and `import tensorflow_probability as tfp`."
```

Comment:
A few minor nit-pics:
1. You can remove the initial "Take note that"
2. In the second sentence, remove "newly added" (since people may read this much later).
3. Missing 't' in inverted. ... Keras considers the inverted mask, ...

Reply:
Good points, pushed a commit to fix these changes. Let me know if there are any other suggestions, thanks!